documentation/general/decouple-vs-and-net-sdk.md

```diff
@@ -0,0 +1,180 @@
+
+Generally this mixing of components is fine because Visual Studio will install a .NET SDK that is functionally paired with it. For example 17.10 installs .NET SDK 8.0.3xx, 17.9 installs .NET SDK 8.0.2xx, etc ... In that scenario the components are the same version and the mixing is largely benign. However when the customer is in a torn state, that is the .NET SDK used is not paired with the Visual Studio version, then compatibility issues can, and **will**, arise. Particularly when the .NET SDK is _newer_ than Visual Studio customers will end up with the following style of error:
+
+> CSC : warning CS9057: The analyzer assembly '..\dotnet\sdk\8.0.200\Sdks\Microsoft.NET.Sdk.Razor\source-
+> generators\Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll' references version '4.9.0.0' of the compiler, which is newer
+> than the currently running version '4.8.0.0'.
+
+These bugs slip by because even though a given .NET SDK is supported in many Visual Studio versions, we do not fully validate its correctness. Specifically, consider that when releasing the .NET 8.0.400 SDK. It is supported in 17.8-17.11 but only fully validated in 17.11. Likewise, when Visual Studio releases servicing fixes it does not attempt to fully validate all supported .NET SDKs. This work on both sides would be prohibitively expensive. Instead, we seek to decouple these experiences so that such testing is not necessary.
+
+Customers existing in a torn state is a common scenario:
+
+|Visual Studio Version | Match% | Float % | Torn % |
+| --- | --- | --- | --- |
+| 16.11 | 81.4% | 10.8% | 7.8% |
+| 17.2 | 91% | 8% | 1% |
+| 17.4 | 91.5% | 5.7% | 2.7% |
+| 17.6 | 91.6% | 5% | 3.5% |
+| 17.7 | 94.9% | 2.2% | 3% |
+| 17.8 | 96% | 0% | 4% |
+
+To address this issue we are going to separate Visual Studio and the .NET SDK in build scenarios. That will change our matrix to be the following:
+
+| Scenario | Loads Roslyn | Loads Analyzers / Generators |
+| --- | --- | --- |
+| msbuild | From .NET SDK | From .NET SDK |
+| dotnet build | From .NET SDK | From .NET SDK |
+| Visual Studio Design Time | From Visual Studio | From Visual Studio |
+
+Specifically we will be:
+
+1. Changing `msbuild` to use tools and tasks from the .NET SDK when building .NET SDK projects.
+2. Changing Visual Studio to use analyzers installed with Visual Studio.
+
+In addition to making our builds more reliable this will also massively simplify our [analyzer Development strategy][sdk-lifecycle]. Analyzers in the SDK following this model can always target the latest Roslyn version without the need for complicated multi-targeting. Further it will allow Visual Studio the ability to NGEN or R2R analyzers which is a long standing request from the Visual Studio perf team.
```

Comment:
```suggestion
In addition to making our builds more reliable this will also massively simplify our [analyzer development strategy][sdk-lifecycle]. Analyzers in the SDK following this model can always target the latest Roslyn version without the need for complicated multi-targeting. Further it will allow Visual Studio the ability to NGEN or R2R analyzers which is a long standing request from the Visual Studio perf team.
```